services/vision/obstaclesdepth/obstacles_depth.go

```diff
@@ -209,24 +211,39 @@ func (o *obsDepth) obsDepthWithIntrinsics(ctx context.Context, src camera.VideoS
 	w, h := dm.Width(), dm.Height()
 	o.dm = dm
 
-	obstaclePoints := make([]image.Point, 0, w*h/sampleN)
+	var wg sync.WaitGroup
+	obstaclePointChan := make(chan image.Point)
+
 	for i := 0; i < w; i += sampleN {
-		for j := 0; j < h; j++ {
-			candidate := image.Pt(i, j)
-		obs: // for every sub-sampled point, figure out if it is an obstacle
-			for l := 0; l < w; l += sampleN { // continue with the sub-sampling
-				for m := 0; m < h; m++ {
-					compareTo := image.Pt(l, m)
-					if candidate == compareTo {
-						continue
-					}
-					if o.isCompatible(candidate, compareTo) {
-						obstaclePoints = append(obstaclePoints, candidate)
-						break obs
+		wg.Add(1)
+		go func(i int) {
```

Comment:
we usually use ManagedGo for these routines

Reply:
I don't think I can only because I'm passing something in (i) to my goroutine. And if I don't make this a funciton of _i_, it'll actually break

Reply:
ah, I see now